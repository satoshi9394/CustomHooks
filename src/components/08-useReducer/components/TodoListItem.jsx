import React from 'react'
import PropTypes from 'prop-types'

const TodoListItem = ({todo, index, handleDelete, handleToggle }) => {
  return (
    <li 
      key={ todo.id }
      className="list-group-item"
    >
      <p 
        className={ `${ todo.done && 'complete' }` }
        onClick={ () => { handleToggle(todo.id) }}
      >
        { index + 1}.-{ todo.desc }
      </p>
      <button
        className="btn btn-danger"
        onClick= { () => { handleDelete(todo.id)}}
      >
        borrar
      </button>
    </li>
  )
}

TodoListItem.propTypes = {
  todo: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
  handleToggle: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired
}

export default TodoListItem
