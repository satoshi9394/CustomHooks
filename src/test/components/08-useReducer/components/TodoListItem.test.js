import { shallow } from 'enzyme';
import React from 'react';
import TodoListItem from "../../../../components/08-useReducer/components/TodoListItem";
import { demoTodos } from '../../../fixtures/demoTodos';

describe('Pruebas en el <TodoListItem/>', () => {

  const handleDelete = jest.fn()
  const handleToggle = jest.fn()

  const wrapper = shallow( <TodoListItem
    todo={ demoTodos[0] }
    index={ 0 }
    handleDelete = { handleDelete }
    handleToggle = { handleToggle }
  />);

  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de llamar a la handleDelete', () => {
    wrapper.find('button').simulate('click');
    expect( handleDelete ).toHaveBeenCalledWith( demoTodos[0].id );
  });

  test('debe de llamar a la handleToggle', () => {
    wrapper.find('p').simulate('click');
    expect( handleToggle ).toHaveBeenCalledWith( demoTodos[0].id );
  });

  test('debe de mostrar el texto correctamente', () => {
    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe( `${ 0 + 1}.-${ demoTodos[0].desc }` );
  });

  test('debe de tener la clase complete si el Todo.done = true', () => {
    const todo = demoTodos[0];
    todo.done = true;
    const wrapper = shallow( <TodoListItem
      todo={ todo }
      index={ 0 }
      handleDelete = { handleDelete }
      handleToggle = { handleToggle }
    />);
    
    expect( wrapper.find('p').hasClass('complete')).toBe(true)
  });

})