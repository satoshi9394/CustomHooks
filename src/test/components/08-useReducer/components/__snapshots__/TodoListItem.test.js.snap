// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <TodoListItem/> debe mostrarse correctamente 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    .-
    Aprender react
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    borrar
  </button>
</li>
`;
